### ISSUE TITLE
Can't render bytes of an image into a variable

### ISSUE THREAD

**Issue by madrobby** [ariya/phantomjs#12084]
`page.render` saves a file, but I'd like to render just the bytes of the rendered image into a variable. I don't want to use base64. Right now, I'm doing…

``` coffeescript
page.render 'temp.png'
image = fs.read 'temp.png', 'b'
fs.remove 'temp.png'
```

…but this is ridiculous because it could be done just in memory. This also causes issues on deployment (needs write rights to directory, etc.).

Could you add an API to render the binary data of a rendered webpage image and just return this byte array (as a string)?

**Comment by MichielDeMey** [comment]
+1

**Comment by peakji** [comment]
@madrobby I've just created a pull request https://github.com/ariya/phantomjs/pull/13345 to provide exacty what you need.

**Comment by madrobby** [comment]
@peakji that looks great!

**Comment by dhendo** [comment]
Closes #11835 as well?

**Comment by zackw** [comment]
I think we could squeeze this into 2.0.1 but please see my comments on the pull request.
